NeosModLoader/Utility/GenericTypeMethodsInvoker.cs

```diff
@@ -0,0 +1,166 @@
+using HarmonyLib;
+using NeosModLoader;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NeosModLoader.Utility
+{
+	internal sealed class GenericTypeMethodsInvoker
+	{
+		private readonly Dictionary<TypeDefinition, ConcreteType> concreteTypes = new();
```

Comment:
why not use a hashset?

Reply:
Why would you use a HashSet to store a relation from basically Type[] to Type?

Nvm that a Dictionary basically is a funny hash set.